tests/test_extra_files.py

```diff
@@ -48,6 +49,150 @@ def test_slashes(test_case):
+    '/http:/': {'http:': {'index.html': b'a'}},
+    '/https:/': {'https:': {'index.html': b'a'}},
+    '/https://': {'https:': {'index.html': b'a'}},
+    '/http:/https:/': {'http:': {'https:': {'index.html': b'a'}}},
+    '/http:/foo/': {'http:': {'foo': {'index.html': b'a'}}},
+}
+@pytest.mark.parametrize('test_case', EXTRA_FILE_DOT)
+def test_dots(test_case):
+    extra_file = {test_case: 'a'}
+    expected = EXTRA_FILE_DOT[test_case]
+    config = {
+        'extra_files': extra_file,
+        'output': {'type': 'dict'},
+    }
+
+    with context_for_test('app_simple') as module:
+        result = freeze(module.app, config)
+
+    # pop to simplify syntax of expected dict
+    # index.html is root page for app_simple, not useful for this test
+    result.pop('index.html')
+
+    assert result == expected
+
+
+EXTRA_FILE_QUOTED = {
+    '%C3%A1/':  {'á': {'index.html': b'a'}},
+    'á/':       {'á': {'index.html': b'a'}},
+    '/a%5cb/':  {'a\\b': {'index.html': b'a'}},
+}
+@pytest.mark.parametrize('test_case', EXTRA_FILE_QUOTED)
+def test_quoted_url_path(test_case):
+    extra_file = {test_case: 'a'}
+    expected = EXTRA_FILE_QUOTED[test_case]
+    config = {
+        'extra_files': extra_file,
+        'output': {'type': 'dict'},
+    }
+
+    with context_for_test('app_simple') as module:
+        result = freeze(module.app, config)
+
+    # pop to simplify syntax of expected dict
+    # index.html is root page for app_simple, not useful for this test
+    result.pop('index.html')
+
+    assert result == expected
+
+EXTRA_FILE_WITH_PREFIX = {
+    # url_part - from configuration: expected url_part - after clean
+    ':':                ':',
+    ':/':               ':/',
+    '.nojekyll':        '.nojekyll',
+    '/.nojekyll':       '.nojekyll',
+    'http:/info.txt':   'http:/info.txt',
+    '/./a/./a.b.':      'a/a.b.',
+    '/./a/./a.b./.':    'a/a.b./',
+    './a/./a.b./.':     'a/a.b./',
+    './a/./a.b././':    'a/a.b./',
+    '/./././a.b./.':    'a.b./',
+    'a..b/a../b..':     'a..b/a../b..',
+    'a..b/a../..b/':    'a..b/a../..b/',
+    '/..b../':          '..b../',
+    '/..b..':           '..b..',
+    'http':             'http',
+    '/a%5cb/':          'a\\b/',
+    '%C3%A1/':          'á/',
+    'á/':               'á/',
+    # http: without leading forward slash alter prefix by werkzeug.url_parse(<prefix>).join("http:")
+    # http: become the protocol of prefix instead url path
+    'http:':            'http:',
+    '/http:':           'http:',
+    '/http:/':          'http:/',
+    '/https:/':         'https:/',
+    '/https://':        'https:/',
+    '/http:/https/':    'http:/https/',
+    r'/http\https/':    'http/https/',
+}
+@pytest.mark.parametrize('url_part', EXTRA_FILE_WITH_PREFIX)
+@pytest.mark.parametrize('prefix', ('https://example.com:443/freezeyt/', 'https://example.com:443/'))
+def test_join_with_prefix(url_part, prefix):
+    if sys.platform.startswith("win") and url_part == '/a%5cb/':
```

Comment:
I am not sure if it's better to use [sys.platform](https://docs.python.org/3/library/sys.html#sys.platform) or [platform.system()](https://docs.python.org/3/library/platform.html#platform.system).

Maybe it's not big deal

Reply:
Try `os.sep`.